web/form.py

```diff
@@ -208,21 +300,58 @@ class Textarea(Input):
     """Textarea input.
     
         >>> Textarea(name='foo', value='bar').render()
-        '<textarea id="foo" name="foo">bar</textarea>'
+        u'<textarea id="foo" name="foo">bar</textarea>'
     """
     def render(self):
         attrs = self.attrs.copy()
         attrs['name'] = self.name
         value = net.websafe(self.value or '')
         return '<textarea %s>%s</textarea>' % (attrs, value)
 
+
+class Datalist(Input):
+    """Datalist input.
+  
+    This is currently supported by only Firefox and Opera. Use it with caucious.
+    Datalist cannot be used separately. It must be bound to an input.
+  
+    An example can be viewed on
+    <http://dev.w3.org/html5/spec/Overview.html#url-state>
+  
+    >>> Datalist(name='list', args=[('a', 'b'), ('c', 'd')]).render()
+    u'<datalist id="list" name="list"><option label="a" value="b"><option label="c" value="d"></datalist>'
+    >>> Datalist(name='list', args=[['a', 'b'], ['c', 'd']]).render()
+    u'<datalist id="list" name="list"><option label="a" value="b"><option label="c" value="d"></datalist>'
+    >>> Datalist(name='list', args=['a', 'b', 'c', 'd']).render()
+    u'<datalist id="list" name="list"><option value="a"><option value="b"><option value="c"><option value="d"></datalist>'
+    """
+    def __init__(self, name, args, *validators, **kwargs):
+        self.args = args
+        super(Datalist, self).__init__(name, *validators, **kwargs)
+  
+    def render(self):
+        attrs = self.attrs.copy()
+        attrs['name'] = self.name
+        label_p = ''
+        x = '<datalist %s>' % attrs
+        for arg in self.args:
+            if isinstance(arg, (tuple, list)):
+                label_p = ' label="%s"' % net.websafe(arg[0])
+                label = net.websafe(arg[1])
+            else:
+                label = net.websafe(arg)
+            x += '<option%s value="%s">' % (label_p, label)
```

Comment:
Suggest to close the option tag:
`            x += '<option%s value="%s" />' % (label_p, label)`